docs/docs/user/references/components.md

```diff
@@ -0,0 +1,245 @@
+# Components
+
+This section explains the different components of KPOps, their usage and
+configuration. To learn more about their hierarchy, visit the
+[architecture](/docs/developer/architecture/component-inheritance.md) page.
```

Comment:
Page does not exist (yet?)

Reply:
Nope, but I wanted to add the reference. It already exists as a PR in the old repo, but I don't think it's coming very soon.

Reply:
I think we should remove/rephrase this broken section

Reply:
I commented the "learn more" sentence. Could be added back when the other page is created.

Reply:
done